lib/iris/tests/unit/fileformats/netcdf/test_Saver__ugrid.py

```diff
@@ -0,0 +1,1171 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Unit tests for the `iris.fileformats.netcdf.Saver` class."""
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests  # isort:skip
+
+from pathlib import Path
+import shutil
+import tempfile
+
+import netCDF4 as nc
+import numpy as np
+
+from iris import save
+from iris.coords import AuxCoord
+from iris.cube import Cube, CubeList
+from iris.experimental.ugrid import Connectivity, Mesh
+from iris.tests.stock import realistic_4d
+
+XY_LOCS = ("x", "y")
+XY_NAMES = ("longitude", "latitude")
+
+
+def build_mesh(
+    n_nodes=2,
+    n_faces=0,
+    n_edges=0,
+    nodecoord_xyargs=None,
+    edgecoord_xyargs=None,
+    facecoord_xyargs=None,
+    conn_role_kwargs=None,  # mapping {connectivity-role: connectivity-kwargs}
+    mesh_kwargs=None,
+):
+    """
+    Make a test mesh.
+
+    Mesh has faces edges, face-coords and edge-coords, numbers of which can be
+    controlled.
+
+    Args:
+
+    * n_nodes, n_faces, n_edges (int):
+        Basic dimensions of mesh components.  Zero means no such location.
+    * nodecoord_xyargs, edgecoord_xyargs, facecoord_xyargs (pair of dict):
+        Pairs (x,y) of settings kwargs, applied after initial creation the
+        relevant location coordinates.
+    * conn_role_kwargs (dict of string:dict):
+        Mapping from cf_role name to settings kwargs for connectivities,
+        applied after initially creating them.
+    * mesh_kwargs (dict):
+        Dictionary of key settings to apply to the Mesh, after creating it.
+
+    """
+
+    def applyargs(coord, kwargs):
+        if kwargs:
+            for key, val in kwargs.items():
+                # kwargs is a dict
+                setattr(coord, key, val)
+
+    def apply_xyargs(coords, xyargs):
+        if xyargs:
+            for coord, kwargs in zip(coords, xyargs):
+                # coords and xyargs both iterables : implicitly=(x,y)
+                applyargs(coord, kwargs)
+
+    node_coords = [
+        AuxCoord(np.arange(n_nodes), standard_name=name)
+        for loc, name in zip(XY_LOCS, XY_NAMES)
```

Comment:
@pp-mo `loc` (`XY_LOCS`) isn't used

Reply:
:+1: nice spot.
I'm surprised flake8 didn't find this though.